packages/NodeCollector/BinaryOpTreeRootLocator.php

```diff
@@ -20,17 +23,24 @@ final class BinaryOpTreeRootLocator
      */
     public function findOperationRoot(Expr $expr, string $binaryOpClass): Expr
     {
-        /** @var ?Expr $parentNode */
         $parentNode = $expr->getAttribute(AttributeKey::PARENT_NODE);
-        if ($parentNode === null || $parentNode::class !== $binaryOpClass) {
+        if (! $parentNode instanceof Expr) {
```

Comment:
same with https://github.com/rectorphp/rector-src/pull/1621

```suggestion
        if (! $parentNode instanceof \PhpParser\Node) {
```